lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/UploaderToJobfolder.java

```diff
@@ -54,8 +56,10 @@
 
   @Inject
   UploaderToJobFolder(@Parameter(JobSubmissionDirectory.class) final String jobSubmissionDirectory,
-                      final YarnConfiguration yarnConfiguration) throws IOException {
+                      final YarnConfiguration yarnConfiguration,
+                      @Parameter(FileSystemUrl.class) final String fileSystemUrl) throws IOException {
     this.jobSubmissionDirectory = jobSubmissionDirectory;
+    YarnConfigurationSetUp.setFileSystemUrl(yarnConfiguration, fileSystemUrl);
```

Comment:
Same as above: now `UploaderToJobFolder` has a side effect modifying `YarnConfiguration`. Remove that call here, update `YarnConfiguration` once immediately after instantiation.